src/TypeScript.UnitTests/Analyzer/JavaScriptAnalyzerTests.cs

```diff
@@ -0,0 +1,298 @@
+            var testSubject = CreateTestSubject(eslintBridgeClient.Object);
+
+            var cancellationToken = new CancellationToken();
+            await testSubject.ExecuteAnalysis("some path", Mock.Of<IIssueConsumer>(), cancellationToken);
+
+            eslintBridgeClient.Verify(x=> x.AnalyzeJs("some path", cancellationToken), Times.Once);
+            eslintBridgeClient.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public async Task ExecuteAnalysis_NoResponse_ConsumerNotCalled()
+        {
+            var eslintBridgeClient = SetupEslintBridgeClient(response: null);
+            var consumer = new Mock<IIssueConsumer>();
+
+            var testSubject = CreateTestSubject(eslintBridgeClient.Object);
+            await testSubject.ExecuteAnalysis("some path", consumer.Object, CancellationToken.None);
+
+            consumer.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public async Task ExecuteAnalysis_ResponseWithNoIssues_ConsumerNotCalled()
+        {
+            var response = new AnalysisResponse();
+            var eslintBridgeClient = SetupEslintBridgeClient(response: response);
+            var consumer = new Mock<IIssueConsumer>();
+
+            var testSubject = CreateTestSubject(eslintBridgeClient.Object);
+            await testSubject.ExecuteAnalysis("some path", consumer.Object, CancellationToken.None);
+
+            consumer.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public async Task ExecuteAnalysis_ResponseWithParsingError_MissingTypescript_ParsingErrorLogged()
+        {
+            var logger = new TestLogger();
+
+            var parsingError = new ParsingError
+            {
+                Code = ParsingErrorCode.MISSING_TYPESCRIPT,
+                Line = 5,
+                Message = "some message"
+            };
+
+            await SetupAnalysisWithParsingError(parsingError, logger);
+
+            logger.AssertPartialOutputStringExists(TypeScript.Analyzer.Resources.ERR_ParsingError_MissingTypescript);
+        }
+
+        [TestMethod]
+        public async Task ExecuteAnalysis_ResponseWithParsingError_UnsupportedTypescript_ParsingErrorLogged()
+        {
+            var logger = new TestLogger();
+
+            var parsingError = new ParsingError
+            {
+                Code = ParsingErrorCode.UNSUPPORTED_TYPESCRIPT,
+                Line = 5,
+                Message = "some message"
+            };
+
+            await SetupAnalysisWithParsingError(parsingError, logger);
+
+            logger.AssertPartialOutputStringExists(TypeScript.Analyzer.Resources.ERR_ParsingError_UnsupportedTypescript);
+        }
+
+        [TestMethod]
+        [DataRow(ParsingErrorCode.FAILING_TYPESCRIPT)]
+        [DataRow(ParsingErrorCode.GENERAL_ERROR)]
+        [DataRow(ParsingErrorCode.PARSING)]
+        [DataRow((ParsingErrorCode)1234)]
+        public async Task ExecuteAnalysis_ResponseWithParsingError_ParsingErrorLogged(int errorCode)
```

Comment:
I can't receive ParsingErrorCode because it's internal; if I change the method to internal, it's not being executed, and if I change the enum to public, it changes a bunch of the contract classes. So I just get it as int.